flocker/apiclient/_client.py

```diff
@@ -111,34 +111,51 @@ class Lease(PClass):
     expires = field(type=(float, int, NoneType), mandatory=True)
 
 
+class MountedDataset(PClass):
+    """
+    A mounted dataset.
+
+    :attr UUID dataset_id: The UUID of the dataset.
+    :attr unicode mountpoint: The filesystem location of the dataset.
+    """
+    dataset_id = field(type=UUID, mandatory=True)
+    mountpoint = field(type=unicode, mandatory=True)
+
+
 class Container(PClass):
     """
     A container in the configuration.
 
-    :attr UUID node_uuid: The UUID of a node in the cluster where the container
-        will run.
+    :attr UUID node_uuid: The UUID of a node in the cluster where the
+        container will run.
     :attr unicode name: The unique name of the container.
     :attr DockerImage image: The Docker image the container will run.
+    :attr Optional[Sequence[MountedDataset]] volumes: Flocker volumes
+        mounted in container.
     """
     node_uuid = field(type=UUID, mandatory=True)
     name = field(type=unicode, mandatory=True)
     image = field(type=DockerImage, mandatory=True)
+    volumes = field(initial=None)
 
 
 class ContainerState(PClass):
     """
     The state of a container in the cluster.
 
-    :attr UUID node_uuid: The UUID of a node in the cluster where the container
-        will run.
+    :attr UUID node_uuid: The UUID of a node in the cluster where the
+        container will run.
     :attr unicode name: The unique name of the container.
     :attr DockerImage image: The name of the Docker image.
     :attr bool running: Whether the container is running.
+    :attr Optional[Sequence[MountedDataset]] volumes: Flocker volumes
+        mounted in container.
     """
     node_uuid = field(type=UUID, mandatory=True)
     name = field(type=unicode, mandatory=True)
     image = field(type=DockerImage, mandatory=True)
     running = field(type=bool, mandatory=True)
+    volumes = field(initial=None)
```

Comment:
Missing `mandatory=True` means that some instances of this might not have the `volumes` attribute set at all (i.e. `state.volumes` will raise an `AttributeError`). Is this deliberate?